extensions/amp-render/1.0/amp-render.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from './base-element';
+// import {CSS} from '../../../build/amp-fit-text-1.0.css';
+import {batchFetchJsonFor} from '../../../src/batched-json';
+import {dict} from '../../../src/utils/object';
+// import {isExperimentOn} from '../../../src/experiments';
+import {Services} from '../../../src/services';
+import {dev, user, userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-render';
+
+const AMP_STATE_URI_SCHEME = 'amp-state:';
+const AMP_SCRIPT_URI_SCHEME = 'amp-script:';
+
+class AmpRender extends BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/template-impl.Templates} */
+    this.templates_ = null;
+
+    /** @private {?Element} */
+    this.template_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    // userAssert(
+    //   isExperimentOn(this.win, 'amp-render'),
+    //   'Experiment "amp-render" is not turned on.'
+    // );
+    return super.isLayoutSupported(layout);
+  }
+
+  /** @override */
+  init() {
+    return dict({
+      'fetchFn': this.getFetchFn_(),
```

Comment:
```suggestion
      'fetchFn': this.getFetchFn_,
```

Reply:
I think you misunderstood. `this.getFetchFn` looks at src to determine if its amp-state, amp-script, etc. & returns the appropriate function to perform the fetch.

Reply:
Ooooh you're right, thanks for clarifying. Sorry I misread!

Reply:
Ooooh you're right, thanks! Will you have to add a mutation observer to `src` to make sure you switch the `fetchFn` if the src changes?

Reply:
are we moving off of mutatedAttributeCallback?

Reply:
It should just work with `mutationObserverCallback`, since the bound `[src]` operation will mutate the `src` attribute with the updated value.

I don't think we can rely on just `BaseElement['props']`'s configuration of `src` (which already puts a mutation observer on those attrs) since we would want to be able to mutate a different prop `fetchFn` based on `src` mutating... but I wonder if we can do something like the following, assuming we can make `getFetchFn` a top level fn that takes an element instead of inside `AmpRender`:
```
BaseElement['props'] = {
  'src': { attr: 'src' },
  'fetchFn': { attrs: ['src'], parseAttrs: (e) => getFetchFn(e)
}
```

Reply:
If such a thing is possible, we can look into moving `getFetchFn` at the top level.

Reply:
This PR is already pretty large. I'll make these changes in a follow up PR.